miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -34,16 +39,10 @@ end
 #! - The fungible asset is not found in the vault.
 #! - The amount of the fungible asset in the vault is less than the amount to be removed.
 #! - The non-fungible asset is not found in the vault.
-export.send_asset.1
+export.send_asset.2
     exec.account::remove_asset
-    # => [ASSET, tag, RECIPIENT, ...]
-
-    # insert 8 ZEROs into the stack right after recipient; we temporarily store one of the
-    # elements of ASSET in memory to make stack manipulation easier
-    push.0 swap loc_store.0 padw push.0.0.0 swapdw loc_load.0
-    # => [ASSET, tag, RECIPIENT, ZERO, ZERO, ...]
-
+    # => [ASSET, tag, note_type, RECIPIENT, ...]
 
-    exec.tx::create_note
-    # => [note_ptr, ZERO, ZERO, ...]
+    syscall.create_note
+    # => [note_ptr, ZERO, ZERO, 0, ...]
```

Comment:
We've tried to avoid using `syscall`'s from "user" code (i.e., something that a user might use to write a smart contract). Instead, all interactions with the kernel would go through the wrappers exposed via various other modules.

One of the reasons for this is so that users don't need to know which instruction to use (i.e., one always uses `exec`). The other one is so that the user does not need to be aware of `syscall`-specific semantics (this one is not applicable in this specific case).

Is there a reason to deviate from this approach here?

Reply:
changed